### PR TITLE
protect bstar in denominator

### PR DESCRIPTION
**Pull request by jiahe23**
# PULL REQUEST

## Purpose and Content
(Clearly and concisely state the purpose of this PR)

## Benefits and Risks
(State concisely the benefits to be derived from and the risks associated with this PR)

## Linked Issues
(Provide references to any link issues. Use closes #issuenum to automatically close an open issue)
- Fixes #
- Closes #

## PR Checklist
- [ ] This PR has a corresponding issue OR is linked to an SDI.
- [ ] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [ ] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [ ] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [ ] I linted my code on my local machine prior to submission OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code used in an integration test OR N/A.
- [ ] All tests ran successfully on my local machine OR N/A.
- [ ] All classes, modules, and function contain docstrings OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/SurfaceFluxes.jl [hunk 1]
```diff
@@ -344,11 +344,11 @@ function obukhov_length end
 
 obukhov_length(sfc::SurfaceFluxConditions) = sfc.L_MO
 
-function non_zero_lmo(L_MO::FT) where {FT}
-    if L_MO == FT(0)
-        return L_MO + sqrt(eps(FT))
+function non_zero(value::FT) where {FT}
+    if value == FT(0)
```
Comment (charleskawczynski) [inline src/SurfaceFluxes.jl:348]: Just to note, we should probably replace this with a numerically better comparison. For example:

```julia
julia> Float64(1e-304) == Float64(0)
false
```
I thought @tapios suggested to use `abs(value) < eps(FT)` or `abs(value) < sqrt(eps(FT))`? This is probably more numerically sensible in terms of avoiding division by zero.

Comment (jiahe23) [reply src/SurfaceFluxes.jl:348]: Actually, it also get protected...
```
julia> non_zero(1e-304)
1.4901161193847656e-8
```

Comment (jiahe23) [reply src/SurfaceFluxes.jl:348]: Or, are you suggesting something like
```
function non_zero(value::FT) where {FT}
           if abs(value) < eps(FT)
               return value + sqrt(eps(FT))
           else
               return value 
           end
       end
```

Comment (charleskawczynski) [reply src/SurfaceFluxes.jl:348]: Ah, good point. I didn't realize that we're just always adding eps in the else case. Yeah, what you wrote is what I was suggesting.

File: test/runtests.jl [hunk 2]
```diff
@@ -140,10 +140,10 @@ const sf_params = SurfaceFluxes.Parameters.SurfaceFluxesParameters{
 
             sfc_output = SF.surface_conditions(sf_params, sc)
             @test SF.obukhov_length(sfc_output) > FT(0)
-            @test sign(SF.non_zero_lmo(1.0)) == 1
-            @test sign(SF.non_zero_lmo(-1.0)) == -1
-            @test sign(SF.non_zero_lmo(-0.0)) == 1
-            @test sign(SF.non_zero_lmo(0.0)) == 1
+            @test sign(SF.non_zero(1.0)) == 1
```
Comment (yairchn) [inline test/runtests.jl:143]: I ma not sure these are good values for `b_star` (1,-1) they might be a bit large

Comment (jiahe23) [reply test/runtests.jl:143]: This is not for bstar. It is to test the `non_zero` function in general. It was called `non_zero_lmo` before when we only protect lmo then. I believe these lines are to test the sign but yes we could add some more physical values.

**Pull request merged**